apps/wallet/src/ui/app/hooks/useGetNFTMeta.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+    getObjectFields,
+    isSuiObject,
+    type GetObjectDataResponse,
+} from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
+import { useMemo } from 'react';
+
+import { api } from '../redux/store/thunk-extras';
```

Comment:
also we should move this to a hook. useApi / useRpc and avoid including it like this in other places